# Estimation And Confidence Intervals

A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows:
1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29
## A. Build 99% Confidence Interval Using Sample Standard Deviation

In [18]:
# import libraries
import pandas as pd
import numpy as np
import scipy.stats as stats
import math
import warnings
warnings.filterwarnings('ignore')

In [24]:
# Durability data in millions of characters
data = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]
# Step 1: Calculate the sample mean
sample_mean = np.mean(data)
# Step 2: Calculate the sample standard deviation. ddof=1 tells the function to divide by n−1 instead of 𝑛.
sample_std = np.std(data,ddof=1)
sample_mean,sample_std

(1.2386666666666666, 0.19316412956959936)

**Since the sample size is small (n = 15) and the population standard deviation is unknown, we'll use the t-distribution. The degrees of freedom (df) for this distribution is 𝑛−1.**

In [25]:
# Step 3: Determine the critical value from the t-distribution for 99% confidence level
n = len(data)
df = n - 1
confidence_level = 0.99
alpha = 1 - confidence_level
t_critical = stats.t.ppf(1 - alpha/2, df)
t_critical

2.97684273411266

In [28]:
# Step 4: Find Margin of error(ME)
ME=t_critical*(sample_std/np.sqrt(n))
ME

0.1484693282152996

In [29]:
# 99% Confidence Interval for the Mean Durability (x_bar-ME,x_bar+ME)
# Step 5: Construct the confidence interval
confidence_interval = (sample_mean - ME, sample_mean + ME)
confidence_interval

(1.090197338451367, 1.3871359948819662)

### Interpretation:
**We are 99% confident that the true mean durability of the print-heads lies between approximately 1.09 and 1.39 million characters before failure.**

* The t-distribution was used because the sample size is small (n=15), and the population standard deviation is unknown.
* This distribution is more spread out than the normal distribution, which accounts for the added uncertainty in estimating the population standard deviation from a small sample.

## B. Build 99% Confidence Interval Using Known Population Standard Deviation
If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.

**When the population standard deviation (σ) is known, we can use the z-distribution (normal distribution) instead of the t-distribution to construct the confidence interval.**

In [32]:
# Given data
population_std = 0.2  # million characters
sample_mean  # previously calculated sample mean
n = 15  # sample size
# Determine the critical value from the z-distribution for 99% confidence level
confidence_level = 0.99
alpha = 1 - confidence_level
z_critical = stats.norm.ppf(1 - alpha/2)
z_critical

2.5758293035489004

In [33]:
# Find Margin of error(ME)
ME=z_critical*(population_std/np.sqrt(n))
ME

0.13301525327090588

In [34]:
# Construct the confidence interval
confidence_interval = (sample_mean - ME, sample_mean + ME)
confidence_interval

(1.1056514133957607, 1.3716819199375725)

### Interpretation:
**We are 99% confident that the true mean durability of the print-heads lies between approximately 1.106 and 1.372 million characters before failure, using the known population standard deviation.**